--- courier-authlib-0.59.2.orig/configure	2007-04-06 22:38:47.000000000 +0300
+++ courier-authlib-0.59.2/configure	2019-12-18 17:33:46.451737069 +0200
@@ -23077,8 +23077,11 @@
 
 EOF
 	sleep 5
-
-	if test -f ${vpopmail_home}/etc/lib_deps
+	if test -f /etc/libvpopmail/lib_deps & -f /usr/lib/libvpopmail
+	then
+		CFLAGS="/etc/libvpopmail/lib_deps $CFLAGS"
+		VPOPMAILLIBS="/usr/lib/libvpopmail"
+	else if -f ${vpopmail_home}/etc/lib_deps
 	then
 		CFLAGS="`cat ${vpopmail_home}/etc/inc_deps` $CFLAGS"
 		VPOPMAILLIBS="`cat ${vpopmail_home}/etc/lib_deps`"
