diff -Naur hiawatha-9.11_ori/polarssl/library/asn1parse.c hiawatha-9.11_mod/polarssl/library/asn1parse.c
--- hiawatha-9.11_ori/polarssl/library/asn1parse.c	2014-10-20 13:59:19.000000000 +0200
+++ hiawatha-9.11_mod/polarssl/library/asn1parse.c	2015-02-06 20:07:03.000000000 +0100
@@ -278,6 +278,8 @@
             if( cur->next == NULL )
                 return( POLARSSL_ERR_ASN1_MALLOC_FAILED );
 
+            memset( cur->next, 0, sizeof( asn1_sequence ) );
+
             cur = cur->next;
         }
     }
