diff -Naur qmail-kng-mod2/dkimbase.h qmail-kng-mod3/dkimbase.h
--- qmail-kng-mod2/dkimbase.h	2024-03-06 13:17:41.000000000 +1100
+++ qmail-kng-mod3/dkimbase.h	2024-03-12 15:18:49.233402420 +1100
@@ -26,9 +26,9 @@
 #ifndef DKIMBASE_H
 #define DKIMBASE_H
 
-#include <openssl/evp.h>
-#include <openssl/pem.h>
-#include <openssl/err.h>
+#include <openssl11/openssl/evp.h>
+#include <openssl11/openssl/pem.h>
+#include <openssl11/openssl/err.h>
 
 #define BUFFER_ALLOC_INCREMENT	256
 
diff -Naur qmail-kng-mod2/dkim.c qmail-kng-mod3/dkim.c
--- qmail-kng-mod2/dkim.c	2024-03-06 13:17:41.000000000 +1100
+++ qmail-kng-mod3/dkim.c	2024-03-12 15:18:30.693721301 +1100
@@ -33,7 +33,7 @@
 #include "dns_text.h"
 
 #ifdef HAVE_OPENSSL_EVP_H
-#include <openssl/evp.h>
+#include <openssl11/openssl/evp.h>
 #define DKIM_MALLOC(s)     OPENSSL_malloc(s)
 #define DKIM_REALLOC(s, n) OPENSSL_realloc((s), (n))
 #define DKIM_MFREE(s)      OPENSSL_free(s); s = NULL;
diff -Naur qmail-kng-mod2/dkimverify.cpp qmail-kng-mod3/dkimverify.cpp
--- qmail-kng-mod2/dkimverify.cpp	2024-03-06 13:17:42.000000000 +1100
+++ qmail-kng-mod3/dkimverify.cpp	2024-03-12 15:19:02.927166888 +1100
@@ -27,8 +27,8 @@
 #include <assert.h>
 #include <vector>
 #include <algorithm>
-#include <openssl/ssl.h>
-#include <openssl/err.h>
+#include <openssl11/openssl/ssl.h>
+#include <openssl11/openssl/err.h>
 #include "dkim.h"
 #include "dkimverify.h"
 #include "time_t_size.h"
diff -Naur qmail-kng-mod2/dns_text.c qmail-kng-mod3/dns_text.c
--- qmail-kng-mod2/dns_text.c	2024-03-06 13:17:42.000000000 +1100
+++ qmail-kng-mod3/dns_text.c	2024-03-12 15:19:30.923685349 +1100
@@ -11,7 +11,7 @@
 #include "error.h"
 #include "byte.h"
 #include "alloc.h"
-#include <openssl/evp.h>
+#include <openssl11/openssl/evp.h>
 
 #define MAX_EDNS_RESPONSE_SIZE 65536
 
diff -Naur qmail-kng-mod2/qmail-dkim.c qmail-kng-mod3/qmail-dkim.c
--- qmail-kng-mod2/qmail-dkim.c	2024-03-06 13:17:54.000000000 +1100
+++ qmail-kng-mod3/qmail-dkim.c	2024-03-12 15:19:39.205542901 +1100
@@ -31,7 +31,7 @@
 #include "config.h"
 #endif
 #include "dkim.h"
-#include <openssl/evp.h>
+#include <openssl11/openssl/evp.h>
 
 #define DKIM_MALLOC(n)     OPENSSL_malloc(n)
 #define DKIM_MFREE(s)      OPENSSL_free(s); s = NULL;
diff -Naur qmail-kng-mod2/qmail-remote.c qmail-kng-mod3/qmail-remote.c
--- qmail-kng-mod2/qmail-remote.c	2024-03-07 10:13:46.000000000 +1100
+++ qmail-kng-mod3/qmail-remote.c	2024-03-12 15:19:51.399333035 +1100
@@ -68,7 +68,7 @@
 # include <sys/stat.h>
 # include "tls.h"
 # include "ssl_timeoutio.h"
-# include <openssl/x509v3.h>
+# include <openssl11/openssl/x509v3.h>
 # define EHLO 1
 
 int tls_init();
diff -Naur qmail-kng-mod2/ssl_timeoutio.h qmail-kng-mod3/ssl_timeoutio.h
--- qmail-kng-mod2/ssl_timeoutio.h	2024-03-06 13:18:03.000000000 +1100
+++ qmail-kng-mod3/ssl_timeoutio.h	2024-03-12 15:20:04.828100970 +1100
@@ -1,7 +1,7 @@
 #ifndef SSL_TIMEOUTIO_H
 #define SSL_TIMEOUTIO_H
 
-#include <openssl/ssl.h>
+#include <openssl11/openssl/ssl.h>
 
 /* the version is like this: 0xMNNFFPPS: major minor fix patch status */
 #if OPENSSL_VERSION_NUMBER < 0x00908000L
diff -Naur qmail-kng-mod2/tls.c qmail-kng-mod3/tls.c
--- qmail-kng-mod2/tls.c	2024-03-06 13:18:05.000000000 +1100
+++ qmail-kng-mod3/tls.c	2024-03-12 15:20:16.748894962 +1100
@@ -1,8 +1,8 @@
 #ifdef TLS
 #include "exit.h"
 #include "error.h"
-#include <openssl/ssl.h>
-#include <openssl/err.h>
+#include <openssl11/openssl/ssl.h>
+#include <openssl11/openssl/err.h>
 
 int smtps = 0;
 SSL *ssl = NULL;
diff -Naur qmail-kng-mod2/tls.h qmail-kng-mod3/tls.h
--- qmail-kng-mod2/tls.h	2024-03-06 13:18:05.000000000 +1100
+++ qmail-kng-mod3/tls.h	2024-03-12 15:20:25.541743010 +1100
@@ -1,7 +1,7 @@
 #ifndef TLS_H
 #define TLS_H
 
-#include <openssl/ssl.h>
+#include <openssl11/openssl/ssl.h>
 
 extern int smtps;
 extern SSL *ssl;
