diff -Naur qmail-1.03.orig/Makefile qmail-1.03/Makefile > qmail_kloxong-vpopmail-devel.patch
--- qmail-1.03.orig/Makefile	1998-06-15 13:53:16.000000000 +0300
+++ qmail-1.03/Makefile	2019-12-14 06:05:46.608334811 +0200
@@ -1,5 +1,8 @@
 # Don't edit Makefile! Use conf-* for configuration.
 
+SMTPD_CHKUSER_OBJ=chkuser.o
+VPOPMAIL_LIBS=`head -1 /etc/libvpopmail/lib_deps`
+
 SHELL=/bin/sh
 
 default: it
