--- qmailadmin-1.2.16.orig/configure.in	2011-02-22 16:45:47.000000000 +0200
+++ qmailadmin-1.2.16/configure.in	2019-12-19 20:22:34.546548425 +0200
@@ -71,10 +71,20 @@
     ] )
 AC_SUBST(vpopmaildir)
 
-auth_incs=`cat $vpopmaildir/etc/inc_deps`
-AC_SUBST(auth_incs)
-auth_libs=`cat $vpopmaildir/etc/lib_deps`
-AC_SUBST(auth_libs)
+if test -f /etc/libvpopmail/lib_deps
+	 then
+		echo "Setting qmail & vpopmail toaster library locations"
+		auth_incs=`cat /etc/libvpopmail/inc_deps`
+		AC_SUBST(auth_incs)
+		auth_libs=`cat /etc/libvpopmail/lib_deps`
+		AC_SUBST(auth_libs)	
+ 	else
+		echo "Setting qmail & vpopmail native library locations"
+		auth_incs=`cat $vpopmaildir/etc/inc_deps`
+		AC_SUBST(auth_incs)
+		auth_libs=`cat $vpopmaildir/etc/lib_deps`
+		AC_SUBST(auth_libs)
+fi
 
 htmldir=""
 for f in /home/httpd/html /usr/local/apache/htdocs /usr/local/www/data /var/www/html
