diff -ruN tht-1.2.6/admin/pages/home.php tht-kloxo-1.2.6/admin/pages/home.php
--- tht-1.2.6/admin/pages/home.php	2013-02-02 17:08:08.000000000 +1100
+++ tht-kloxo-1.2.6/admin/pages/home.php	2019-12-19 11:33:39.613000009 +1100
@@ -82,7 +82,7 @@
 			}
 		}
 		$array['NOTEPAD'] = $db->resources('admin_notes');
-		$content_notepad = $style->replaceVar('tpl/notepad.tpl', $array);
+/*		$content_notepad = $style->replaceVar('tpl/notepad.tpl', $array);
 		echo '<br />'; //br it, br it
 		echo $main->table('Admin Notepad', $content_notepad, 'auto', 'auto');
 		require_once(LINK.'rss/rss_fetch.inc');
@@ -106,6 +106,7 @@
 		}
 		echo "<br />";
 		echo $main->table('THT News & Updates', $news);
+*/
 	}
 }
 ?>
diff -ruN tht-1.2.6/includes/csrf-secret.php tht-kloxo-1.2.6/includes/csrf-secret.php
--- tht-1.2.6/includes/csrf-secret.php	1970-01-01 10:00:00.000000000 +1000
+++ tht-kloxo-1.2.6/includes/csrf-secret.php	2019-12-19 11:35:12.643994449 +1100
@@ -0,0 +1 @@
+<?php $secret = "A2A0882792AB3930396DD6972FA4D9322ECD0AFC";
\ No newline at end of file
diff -ruN tht-1.2.6/includes/tpl/addserver.tpl tht-kloxo-1.2.6/includes/tpl/addserver.tpl
--- tht-1.2.6/includes/tpl/addserver.tpl	2011-04-09 18:37:55.000000000 +1000
+++ tht-kloxo-1.2.6/includes/tpl/addserver.tpl	2019-12-19 11:38:35.604000010 +1100
@@ -37,7 +37,7 @@
   </tr>
   <tr>
     <td>Username:</td>
-    <td><input type="text" name="user" id="user" /><a title="Username to connect to WebHost Manager" class="tooltip"><img src="<URL>themes/icons/information.png" /></a></td>
+    <td><input type="text" name="user" id="user" /><a title="Username ('Auxiliary Login' for Kloxo/Kloxo-NG) to connect to WebHost Manager" class="tooltip"><img src="<URL>themes/icons/information.png" /></a></td>
   </tr>
   <tr>
     <td valign="top" width="20%"><span id="passtext"></span></td>
diff -ruN tht-1.2.6/includes/tpl/viewserver.tpl tht-kloxo-1.2.6/includes/tpl/viewserver.tpl
--- tht-1.2.6/includes/tpl/viewserver.tpl	2011-04-09 18:44:25.000000000 +1000
+++ tht-kloxo-1.2.6/includes/tpl/viewserver.tpl	2019-12-19 11:41:14.634000010 +1100
@@ -41,8 +41,8 @@
   <tr>
     <td>Username:</td>
     <td><input type="text" name="user" id="user" value="%USER%" />
-    <a title="The username to access the server." class="tooltip"><img src="<URL>themes/icons/information.png" /></a></td>
-  </tr>
+    <a title="The username ('Auxiliary Login' for Kloxo/Kloxo-NG to access the server." class="tooltip"><img src="<URL>themes/icons/information.png" /></a></td>
+  </tr>
   <tr>
     <td valign="top" width="20%"><span id="passtext"></span></td>
     <td><span id="passbox"></span></td>
