diff -Naur tht-1.2.6_ori/admin/pages/home.php tht-1.2.6_mod/admin/pages/home.php
--- tht-1.2.6_ori/admin/pages/home.php	2013-02-02 07:08:08.000000000 +0100
+++ tht-1.2.6_mod/admin/pages/home.php	2015-04-25 10:47:42.000000000 +0200

@@ -82,7 +82,7 @@ class page {
 			}
 		}
 		$array['NOTEPAD'] = $db->resources('admin_notes');
-		$content_notepad = $style->replaceVar('tpl/notepad.tpl', $array);
+	/*	$content_notepad = $style->replaceVar('tpl/notepad.tpl', $array);
 		echo '<br />'; //br it, br it
 		echo $main->table('Admin Notepad', $content_notepad, 'auto', 'auto');
 		require_once(LINK.'rss/rss_fetch.inc');
@@ -106,6 +106,7 @@ class page {
 		}
 		echo "<br />";
 		echo $main->table('THT News & Updates', $news);
+	*/	
 	}
 }
 ?>
diff -Naur tht-1.2.6_ori/includes/csrf-secret.php tht-1.2.6_mod/includes/csrf-secret.php
--- tht-1.2.6_ori/includes/csrf-secret.php	1970-01-01 01:00:00.000000000 +0100
+++ tht-1.2.6_mod/includes/csrf-secret.php	2015-04-24 15:11:46.000000000 +0200
@@ -0,0 +1 @@
+<?php $secret = "A2A0882792AB3930396DD6972FA4D9322ECD0AFC";
diff -Naur tht-1.2.6_ori/includes/tpl/addserver.tpl tht-1.2.6_mod/includes/tpl/addserver.tpl
--- tht-1.2.6_ori/includes/tpl/addserver.tpl	2011-04-09 10:37:55.000000000 +0200
+++ tht-1.2.6_mod/includes/tpl/addserver.tpl	2015-04-24 15:32:21.000000000 +0200
@@ -37,7 +37,7 @@ $(window).load(function () {
   </tr>
   <tr>
     <td>Username:</td>
-    <td><input type="text" name="user" id="user" /><a title="Username to connect to WebHost Manager" class="tooltip"><img src="<URL>themes/icons/information.png" /></a></td>
+    <td><input type="text" name="user" id="user" /><a title="Username ('Auxiliary Login' for Kloxo/Kloxo-NG) to connect to WebHost Manager" class="tooltip"><img src="<URL>themes/icons/information.png" /></a></td>
   </tr>
   <tr>
     <td valign="top" width="20%"><span id="passtext"></span></td>
diff -Naur tht-1.2.6_ori/includes/tpl/viewserver.tpl tht-1.2.6_mod/includes/tpl/viewserver.tpl
--- tht-1.2.6_ori/includes/tpl/viewserver.tpl	2011-04-09 10:44:25.000000000 +0200
+++ tht-1.2.6_mod/includes/tpl/viewserver.tpl	2015-04-24 15:29:13.000000000 +0200
@@ -41,8 +41,8 @@ $(window).load(function () {
   <tr>
     <td>Username:</td>
     <td><input type="text" name="user" id="user" value="%USER%" />
-    <a title="The username to access the server." class="tooltip"><img src="<URL>themes/icons/information.png" /></a></td>
-  </tr>
+    <a title="The username ('Auxiliary Login' for Kloxo/Kloxo-NG to access the server." class="tooltip"><img src="<URL>themes/icons/information.png" /></a></td>
+	</tr>
   <tr>
     <td valign="top" width="20%"><span id="passtext"></span></td>
     <td><span id="passbox"></span></td>
