diff -Naur pure-ftpd-1.0.40.orig/pam/pure-ftpd pure-ftpd-1.0.40/pam/pure-ftpd
--- pure-ftpd-1.0.40.orig/pam/pure-ftpd	2015-06-13 12:26:37.000000000 +0200
+++ pure-ftpd-1.0.40/pam/pure-ftpd	2015-06-01 16:36:31.000000000 +0200
@@ -4,12 +4,14 @@
 # Install it in /etc/pam.d/pure-ftpd or add to /etc/pam.conf
 
 auth       required     pam_listfile.so item=user sense=deny file=/etc/ftpusers onerr=succeed
+auth       include      password-auth
 auth       required     pam_shells.so
 auth       required     pam_nologin.so
 
-account    required     pam_stack.so service=system-auth
+account    include      password-auth
 
-password   required     pam_stack.so service=system-auth
+password   include      password-auth
 
-session    required     pam_stack.so service=system-auth
+session    required     pam_loginuid.so
+session    include      password-auth
 
