diff -up a/pure-ftpd.conf.in b/pure-ftpd.conf.in
--- a/pure-ftpd.conf.in	2017-09-07 13:47:26.000000000 +0200
+++ b/pure-ftpd.conf.in	2019-02-05 17:27:07.475603616 +0100
@@ -9,7 +9,7 @@
 # instead of command-line options, please run the
 # following command :
 #
-# @prefix@/sbin/pure-ftpd @sysconfdir@/etc/pure-ftpd.conf
+# @sbindir@/pure-ftpd @sysconfdir@/pure-ftpd.conf
 #
 # Online documentation:
 # https://www.pureftpd.org/project/pure-ftpd/doc
@@ -106,34 +106,34 @@
 
 # LDAP configuration file (see README.LDAP)
 
-# LDAPConfigFile               /etc/pureftpd-ldap.conf
+# LDAPConfigFile                @sysconfdir@/pureftpd-ldap.conf
 
 
 
 # MySQL configuration file (see README.MySQL)
 
-# MySQLConfigFile              /etc/pureftpd-mysql.conf
+# MySQLConfigFile               @sysconfdir@/pureftpd-mysql.conf
 
 
 # PostgreSQL configuration file (see README.PGSQL)
 
-# PGSQLConfigFile              /etc/pureftpd-pgsql.conf
+# PGSQLConfigFile               @sysconfdir@/pureftpd-pgsql.conf
 
 
 # PureDB user database (see README.Virtual-Users)
 
-# PureDB                       /etc/pureftpd.pdb
+# PureDB                        @sysconfdir@/pureftpd.pdb
 
 
 # Path to pure-authd socket (see README.Authentication-Modules)
 
-# ExtAuth                      /var/run/ftpd.sock
+# ExtAuth                       @localstatedir@/run/ftpd.sock
 
 
 
 # If you want to enable PAM authentication, uncomment the following line
 
-# PAMAuthentication            yes
+PAMAuthentication             yes
 
 
 
@@ -238,7 +238,14 @@
 # For example, a value of 100 prevents all users whose user id is below
 # 100 from logging in. If you want "root" to be able to log in, use 0.
 
-MinUID                       100
+MinUID                      1000
+
+
+
+# Do not use the /etc/ftpusers file to disable accounts. We're already
+# using MinUID to block users with uid < 1000
+
+UseFtpUsers no
 
 
 
@@ -277,7 +284,7 @@
 
 # Prevent anonymous users from uploading new files (no = upload is allowed)
 
-AnonymousCantUpload          no
+AnonymousCantUpload         yes
 
 
 
@@ -301,21 +308,21 @@
 # fw.c9x.org - jedi [13/Apr/2017:19:36:39] "GET /ftp/linux.tar.bz2" 200 21809338
 # This log file can then be processed by common HTTP traffic analyzers.
 
-# AltLog                       clf:/var/log/pureftpd.log
+AltLog                     clf:@localstatedir@/log/pureftpd.log
 
 
 
 # Create an additional log file with transfers logged in a format optimized
 # for statistic reports.
 
-# AltLog                       stats:/var/log/pureftpd.log
+# AltLog                     stats:@localstatedir@/log/pureftpd.log
 
 
 
 # Create an additional log file with transfers logged in the standard W3C
 # format (compatible with many HTTP log analyzers)
 
-# AltLog                       w3c:/var/log/pureftpd.log
+# AltLog                     w3c:@localstatedir@/log/pureftpd.log
 
 
 
@@ -346,9 +353,9 @@
 
 
 # If your pure-ftpd has been compiled with standalone support, you can change
-# the location of the pid file. The default is /var/run/pure-ftpd.pid
+# the location of the pid file. The default is @localstatedir@/run/pure-ftpd.pid
 
-# PIDFile                      /var/run/pure-ftpd.pid
+#PIDFile                     @localstatedir@/run/pure-ftpd.pid
 
 
 
