From 8b6e9370ccbdff47107963c352caab24fdb0fcaa Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Lyson=C4=9Bk?= <olysonek@redhat.com>
Date: Wed, 7 Feb 2018 13:28:02 +0100
Subject: [PATCH 1/2] Exit when given an invalid cmdline option

---
 src/ftpd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ftpd.c b/src/ftpd.c
index e19d8de..5c77410 100644
--- a/src/ftpd.c
+++ b/src/ftpd.c
@@ -6192,7 +6192,7 @@ int pureftpd_start(int argc, char *argv[], const char *home_directory_)
             exit(EXIT_SUCCESS);
         }
         default:
-            logfile(LOG_WARNING, MSG_ILLEGAL_OPTION);
+            die(421, LOG_ERR, MSG_ILLEGAL_OPTION);
         }
     }
 #ifdef WITH_RFC2640
-- 
2.14.3

