diff -Naur pure-ftpd-1.0.49/configuration-file/Makefile.am pure-ftpd-1.0.49.new/configuration-file/Makefile.am
--- pure-ftpd-1.0.49/configuration-file/Makefile.am	1970-01-01 10:00:00.000000000 +1000
+++ pure-ftpd-1.0.49.new/configuration-file/Makefile.am	2020-02-04 21:18:32.000000000 +1100
@@ -0,0 +1,2 @@
+CLEANFILES = \
+	pure-config.pl
diff -Naur pure-ftpd-1.0.49/configuration-file/pure-config.pl.in pure-ftpd-1.0.49.new/configuration-file/pure-config.pl.in
--- pure-ftpd-1.0.49/configuration-file/pure-config.pl.in	1970-01-01 10:00:00.000000000 +1000
+++ pure-ftpd-1.0.49.new/configuration-file/pure-config.pl.in	2020-02-04 21:12:27.000000000 +1100
@@ -0,0 +1,127 @@
+#! @PERL@
+
+# (C) 2001-2013 Aristotle Pagaltzis
+# derived from code (C) 2001-2002 Frank Denis and Matthias Andree
+
+use strict;
+
+my ($conffile, @flg) = @ARGV;
+
+my $PUREFTPD;
+-x && ($PUREFTPD=$_, last) for qw(
+        ${exec_prefix}/sbin/pure-ftpd
+        /usr/local/pure-ftpd/sbin/pure-ftpd
+        /usr/local/pureftpd/sbin/pure-ftpd
+        /usr/local/sbin/pure-ftpd
+        /usr/sbin/pure-ftpd
+);
+
+my %simple_switch_for = (
+        IPV4Only                        => "-4",
+        IPV6Only                        => "-6",
+        ChrootEveryone                  => "-A",
+        BrokenClientsCompatibility      => "-b",
+        Daemonize                       => "-B",
+        VerboseLog                      => "-d",
+        DisplayDotFiles                 => "-D",
+        AnonymousOnly                   => "-e",
+        NoAnonymous                     => "-E",
+        DontResolve                     => "-H",
+        AnonymousCanCreateDirs          => "-M",
+        NATmode                         => "-N",
+        CallUploadScript                => "-o",
+        AntiWarez                       => "-s",
+        AllowUserFXP                    => "-w",
+        AllowAnonymousFXP               => "-W",
+        ProhibitDotFilesWrite           => "-x",
+        ProhibitDotFilesRead            => "-X",
+        AllowDotFiles                   => "-z",
+        AutoRename                      => "-r",
+        AnonymousCantUpload             => "-i",
+        LogPID                          => "-1",
+        NoChmod                         => "-R",
+        KeepAllFiles                    => "-K",
+        CreateHomeDir                   => "-j",
+        NoRename                        => "-G",
+        CustomerProof                   => "-Z",
+        NoTruncate                      => "-0",
+);
+
+my %string_switch_for = (
+        FileSystemCharset       => "-8",
+        ClientCharset           => "-9",
+        SyslogFacility          => "-f",
+        FortunesFile            => "-F",
+        ForcePassiveIP          => "-P",
+        Bind                    => "-S",
+        AnonymousBandwidth      => "-t",
+        UserBandwidth           => "-T",
+        TrustedIP               => "-V",
+        AltLog                  => "-O",
+        PIDFile                 => "-g",
+        TLSCipherSuite          => "-J",
+        CertFile                => "-2",
+);
+
+my %numeric_switch_for = (
+        MaxIdleTime             => "-I",
+        MaxDiskUsage            => "-k",
+        TrustedGID              => "-a",
+        MaxClientsNumber        => "-c",
+        MaxClientsPerIP         => "-C",
+        MaxLoad                 => "-m",
+        MinUID                  => "-u",
+        TLS                     => "-Y",
+);
+
+my %numpairb_switch_for = (
+        LimitRecursion          => "-L",
+        PassivePortRange        => "-p",
+        AnonymousRatio          => "-q",
+        UserRatio               => "-Q",
+);
+
+my %numpairc_switch_for = (
+        Umask           => "-U",
+        Quota           => "-n",
+        PerUserLimits   => "-y",
+);
+
+my %auth_method_for = (
+        LDAPConfigFile          => "ldap",
+        MySQLConfigFile         => "mysql",
+        PGSQLConfigFile         => "pgsql",
+        PureDB                  => "puredb",
+        ExtAuth                 => "extauth",
+);
+
+my $simple_switch = qr/(@{[join "|", keys %simple_switch_for ]})\s+yes/i;
+my $string_switch = qr/(@{[join "|", keys %string_switch_for ]})\s+(\S+)/i;
+my $numeric_switch = qr/(@{[join "|", keys %numeric_switch_for ]})\s+(\d+)/i;
+my $numpairb_switch = qr/(@{[join "|", keys %numpairb_switch_for ]})\s+(\d+)\s+(\d+)/i;
+my $numpairc_switch = qr/(@{[join "|", keys %numpairc_switch_for ]})\s+(\d+):(\d+)/i;
+my $auth_method = qr/(@{[join "|", keys %auth_method_for ]})\s+(\S+)/i;
+
+die "Usage: pure-config.pl <configuration file> [extra options]\n"
+        unless defined $conffile;
+
+open CONF, "< $conffile" or die "Can't open $conffile: $!\n";
+
+!/^\s*(?:$|#)/ and (chomp, push @flg,
+        /$simple_switch/i               ? ($simple_switch_for{$1}) :
+        /$string_switch/i               ? ($string_switch_for{$1} . $2) :
+        /$numeric_switch/i              ? ($numeric_switch_for{$1} . $2) :
+        /$numpairb_switch/i             ? ($numpairb_switch_for{$1} . "$2:$3") :
+        /$numpairc_switch/i             ? ($numpairc_switch_for{$1} . "$2:$3") :
+        /$auth_method/i                 ? ("-l" . "$auth_method_for{$1}:$2") :
+        /UnixAuthentication\s+yes/i     ? ("-l" . "unix") :
+        /PAMAuthentication\s+yes/i      ? ("-l" . "pam") :
+        ()
+) while <CONF>;
+
+close CONF;
+
+if (-t STDOUT) {
+        print "Running: $PUREFTPD ", join(" ", @flg), "\n";
+}
+exec { $PUREFTPD } ($PUREFTPD, @flg) or die "cannot exec $PUREFTPD: $!";
diff -Naur pure-ftpd-1.0.49/configure.ac pure-ftpd-1.0.49.new/configure.ac
--- pure-ftpd-1.0.49/configure.ac	2019-04-03 21:41:30.000000000 +1100
+++ pure-ftpd-1.0.49.new/configure.ac	2020-02-04 21:23:07.000000000 +1100
@@ -96,6 +96,14 @@
 AX_CHECK_LINK_FLAG([-Wl,-z,now], [LDFLAGS="$LDFLAGS -Wl,-z,now"])
 AX_CHECK_LINK_FLAG([-Wl,-z,noexecstack], [LDFLAGS="$LDFLAGS -Wl,-z,noexecstack"])
 
+AC_ARG_VAR(PERL,local path to the perl interpreter)
+perl_possible_path="/usr/bin:/usr/local/bin:/bin:/opt/perl/bin:/opt/perl/usr/bin:/opt/perl/usr/local/bin"
+AC_PATH_PROG(PERL,perl,/usr/bin/env perl,$perl_possible_path)
+
+if test -d /usr/local/include; then
+    CPPFLAGS="$CPPFLAGS -I/usr/local/include"
+fi
+
 if test "x$cross_compiling" != "xyes"; then
   for path in \
     /usr/kerberos \
@@ -1465,6 +1473,8 @@
 AC_CONFIG_FILES(Makefile src/Makefile pam/Makefile man/Makefile
 gui/Makefile m4/Makefile pure-ftpd.conf puredb/Makefile
 puredb/src/Makefile
+configuration-file/Makefile
+configuration-file/pure-config.pl
 man/pure-ftpd.8 man/pure-ftpwho.8 man/pure-mrtginfo.8 man/pure-uploadscript.8
 man/pure-statsdecode.8 man/pure-quotacheck.8 man/pure-pw.8 man/pure-pwconvert.8
 man/pure-authd.8 man/pure-certd.8)
diff -Naur pure-ftpd-1.0.49/Makefile.am pure-ftpd-1.0.49.new/Makefile.am
--- pure-ftpd-1.0.49/Makefile.am	2019-03-26 05:04:04.000000000 +1100
+++ pure-ftpd-1.0.49.new/Makefile.am	2020-02-04 21:12:27.000000000 +1100
@@ -43,6 +43,7 @@
 	man \
 	pam \
 	gui \
+	configuration-file \
 	m4
 
 install-data-local:
diff -Naur pure-ftpd-1.0.49/pam/pure-ftpd pure-ftpd-1.0.49.new/pam/pure-ftpd
--- pure-ftpd-1.0.49/pam/pure-ftpd	2018-01-04 23:13:36.000000000 +1100
+++ pure-ftpd-1.0.49.new/pam/pure-ftpd	2020-02-04 21:12:27.000000000 +1100
@@ -4,12 +4,13 @@
 # Install it in /etc/pam.d/pure-ftpd or add to /etc/pam.conf
 
 auth       required     pam_listfile.so item=user sense=deny file=/etc/ftpusers onerr=succeed
+auth       include      password-auth
 auth       required     pam_shells.so
 auth       required     pam_nologin.so
 
-account    required     pam_stack.so service=system-auth
+account    include      password-auth
 
-password   required     pam_stack.so service=system-auth
-
-session    required     pam_stack.so service=system-auth
+password   include      password-auth
 
+session    required     pam_loginuid.so
+session    include      password-auth
diff -Naur pure-ftpd-1.0.49/pure-config.pl.in pure-ftpd-1.0.49.new/pure-config.pl.in
--- pure-ftpd-1.0.49/pure-config.pl.in	1970-01-01 10:00:00.000000000 +1000
+++ pure-ftpd-1.0.49.new/pure-config.pl.in	2020-02-04 21:00:54.000000000 +1100
@@ -0,0 +1,127 @@
+#! @PERL@
+
+# (C) 2001-2013 Aristotle Pagaltzis
+# derived from code (C) 2001-2002 Frank Denis and Matthias Andree
+
+use strict;
+
+my ($conffile, @flg) = @ARGV;
+
+my $PUREFTPD;
+-x && ($PUREFTPD=$_, last) for qw(
+        ${exec_prefix}/sbin/pure-ftpd
+        /usr/local/pure-ftpd/sbin/pure-ftpd
+        /usr/local/pureftpd/sbin/pure-ftpd
+        /usr/local/sbin/pure-ftpd
+        /usr/sbin/pure-ftpd
+);
+
+my %simple_switch_for = (
+        IPV4Only                        => "-4",
+        IPV6Only                        => "-6",
+        ChrootEveryone                  => "-A",
+        BrokenClientsCompatibility      => "-b",
+        Daemonize                       => "-B",
+        VerboseLog                      => "-d",
+        DisplayDotFiles                 => "-D",
+        AnonymousOnly                   => "-e",
+        NoAnonymous                     => "-E",
+        DontResolve                     => "-H",
+        AnonymousCanCreateDirs          => "-M",
+        NATmode                         => "-N",
+        CallUploadScript                => "-o",
+        AntiWarez                       => "-s",
+        AllowUserFXP                    => "-w",
+        AllowAnonymousFXP               => "-W",
+        ProhibitDotFilesWrite           => "-x",
+        ProhibitDotFilesRead            => "-X",
+        AllowDotFiles                   => "-z",
+        AutoRename                      => "-r",
+        AnonymousCantUpload             => "-i",
+        LogPID                          => "-1",
+        NoChmod                         => "-R",
+        KeepAllFiles                    => "-K",
+        CreateHomeDir                   => "-j",
+        NoRename                        => "-G",
+        CustomerProof                   => "-Z",
+        NoTruncate                      => "-0",
+);
+
+my %string_switch_for = (
+        FileSystemCharset       => "-8",
+        ClientCharset           => "-9",
+        SyslogFacility          => "-f",
+        FortunesFile            => "-F",
+        ForcePassiveIP          => "-P",
+        Bind                    => "-S",
+        AnonymousBandwidth      => "-t",
+        UserBandwidth           => "-T",
+        TrustedIP               => "-V",
+        AltLog                  => "-O",
+        PIDFile                 => "-g",
+        TLSCipherSuite          => "-J",
+        CertFile                => "-2",
+);
+
+my %numeric_switch_for = (
+        MaxIdleTime             => "-I",
+        MaxDiskUsage            => "-k",
+        TrustedGID              => "-a",
+        MaxClientsNumber        => "-c",
+        MaxClientsPerIP         => "-C",
+        MaxLoad                 => "-m",
+        MinUID                  => "-u",
+        TLS                     => "-Y",
+);
+
+my %numpairb_switch_for = (
+        LimitRecursion          => "-L",
+        PassivePortRange        => "-p",
+        AnonymousRatio          => "-q",
+        UserRatio               => "-Q",
+);
+
+my %numpairc_switch_for = (
+        Umask           => "-U",
+        Quota           => "-n",
+        PerUserLimits   => "-y",
+);
+
+my %auth_method_for = (
+        LDAPConfigFile          => "ldap",
+        MySQLConfigFile         => "mysql",
+        PGSQLConfigFile         => "pgsql",
+        PureDB                  => "puredb",
+        ExtAuth                 => "extauth",
+);
+
+my $simple_switch = qr/(@{[join "|", keys %simple_switch_for ]})\s+yes/i;
+my $string_switch = qr/(@{[join "|", keys %string_switch_for ]})\s+(\S+)/i;
+my $numeric_switch = qr/(@{[join "|", keys %numeric_switch_for ]})\s+(\d+)/i;
+my $numpairb_switch = qr/(@{[join "|", keys %numpairb_switch_for ]})\s+(\d+)\s+(\d+)/i;
+my $numpairc_switch = qr/(@{[join "|", keys %numpairc_switch_for ]})\s+(\d+):(\d+)/i;
+my $auth_method = qr/(@{[join "|", keys %auth_method_for ]})\s+(\S+)/i;
+
+die "Usage: pure-config.pl <configuration file> [extra options]\n"
+        unless defined $conffile;
+
+open CONF, "< $conffile" or die "Can't open $conffile: $!\n";
+
+!/^\s*(?:$|#)/ and (chomp, push @flg,
+        /$simple_switch/i               ? ($simple_switch_for{$1}) :
+        /$string_switch/i               ? ($string_switch_for{$1} . $2) :
+        /$numeric_switch/i              ? ($numeric_switch_for{$1} . $2) :
+        /$numpairb_switch/i             ? ($numpairb_switch_for{$1} . "$2:$3") :
+        /$numpairc_switch/i             ? ($numpairc_switch_for{$1} . "$2:$3") :
+        /$auth_method/i                 ? ("-l" . "$auth_method_for{$1}:$2") :
+        /UnixAuthentication\s+yes/i     ? ("-l" . "unix") :
+        /PAMAuthentication\s+yes/i      ? ("-l" . "pam") :
+        ()
+) while <CONF>;
+
+close CONF;
+
+if (-t STDOUT) {
+        print "Running: $PUREFTPD ", join(" ", @flg), "\n";
+}
+exec { $PUREFTPD } ($PUREFTPD, @flg) or die "cannot exec $PUREFTPD: $!";
diff -Naur pure-ftpd-1.0.49/pure-ftpd.conf.in pure-ftpd-1.0.49.new/pure-ftpd.conf.in
--- pure-ftpd-1.0.49/pure-ftpd.conf.in	2019-03-26 02:48:42.000000000 +1100
+++ pure-ftpd-1.0.49.new/pure-ftpd.conf.in	2020-02-04 21:12:27.000000000 +1100
@@ -9,7 +9,7 @@
 # instead of command-line options, please run the
 # following command :
 #
-# @sbindir@/sbin/pure-ftpd @sysconfdir@/pure-ftpd.conf
+# @sbindir@/pure-ftpd @sysconfdir@/pure-ftpd.conf
 #
 # Online documentation:
 # https://www.pureftpd.org/project/pure-ftpd/doc
@@ -42,7 +42,7 @@
 
 
 # Run as a background process
-
+# Important: this must be set to 'yes' for the systemd service to work.
 Daemonize                    yes
 
 
@@ -106,34 +106,34 @@
 
 # LDAP configuration file (see README.LDAP)
 
-# LDAPConfigFile               /etc/pureftpd-ldap.conf
+# LDAPConfigFile               @sysconfigdir@/pureftpd-ldap.conf
 
 
 
 # MySQL configuration file (see README.MySQL)
 
-# MySQLConfigFile              /etc/pureftpd-mysql.conf
+# MySQLConfigFile              @sysconfigdir@/pureftpd-mysql.conf
 
 
 # PostgreSQL configuration file (see README.PGSQL)
 
-# PGSQLConfigFile              /etc/pureftpd-pgsql.conf
+# PGSQLConfigFile              @sysconfigdir@/pureftpd-pgsql.conf
 
 
 # PureDB user database (see README.Virtual-Users)
 
-# PureDB                       /etc/pureftpd.pdb
+# PureDB                       @sysconfigdir@/pureftpd.pdb
 
 
 # Path to pure-authd socket (see README.Authentication-Modules)
 
-# ExtAuth                      /var/run/ftpd.sock
+# ExtAuth                      @localstatedir@/run/ftpd.sock
 
 
 
 # If you want to enable PAM authentication, uncomment the following line
 
-# PAMAuthentication            yes
+PAMAuthentication            yes
 
 
 
@@ -238,8 +238,7 @@
 # For example, a value of 100 prevents all users whose user id is below
 # 100 from logging in. If you want "root" to be able to log in, use 0.
 
-MinUID                       100
-
+MinUID                       1000
 
 
 # Allow FXP transfers for authenticated users.
@@ -277,7 +276,7 @@
 
 # Prevent anonymous users from uploading new files (no = upload is allowed)
 
-AnonymousCantUpload          no
+AnonymousCantUpload          yes
 
 
 
@@ -301,21 +300,21 @@
 # fw.c9x.org - jedi [13/Apr/2017:19:36:39] "GET /ftp/linux.tar.bz2" 200 21809338
 # This log file can then be processed by common HTTP traffic analyzers.
 
-# AltLog                       clf:/var/log/pureftpd.log
+AltLog                       clf:@localstatedir@/log/pureftpd.log
 
 
 
 # Create an additional log file with transfers logged in a format optimized
 # for statistic reports.
 
-# AltLog                       stats:/var/log/pureftpd.log
+# AltLog                       stats:@localstatedir@/log/pureftpd.log
 
 
 
 # Create an additional log file with transfers logged in the standard W3C
 # format (compatible with many HTTP log analyzers)
 
-# AltLog                       w3c:/var/log/pureftpd.log
+# AltLog                       w3c:@localstatedir@/log/pureftpd.log
 
 
 
@@ -346,9 +345,9 @@
 
 
 # If your pure-ftpd has been compiled with standalone support, you can change
-# the location of the pid file. The default is /var/run/pure-ftpd.pid
+# the location of the pid file. The default is @localstatedir@/run/pure-ftpd.pid
 
-# PIDFile                      /var/run/pure-ftpd.pid
+#PIDFile                     @localstatedir@/run/pure-ftpd.pid
 
 
 
