--- trafficserver-3.2.0/rc/trafficserver.in.orginal	2012-06-23 15:02:25.768405925 +0200
+++ trafficserver-3.2.0/rc/trafficserver.in	2012-06-23 15:07:16.507193268 +0200
@@ -16,12 +16,21 @@
 # limitations under the License.
 #
 # Template from /etc/init.d/skeleton on Ubuntu
+# trafficserver    Startup/shutdown script for the Apache Traffic Server
+#
+# chkconfig: - 90 10
+# description: TApache Traffic Server is fast, scalable and extensible  \
+#             HTTP/1.1 compliant caching proxy server.
+# processname: traffic_cop traffic_manager traffic_server
+# config: /etc/trafficserver
+# config: /etc/sysconfig/trafficserver
+# pidfile: /var/run/trafficserver/cop.pid
 #
 ### BEGIN INIT INFO
 # Provides:          Traffic Server
 # Required-Start:    $local_fs $remote_fs
 # Required-Stop:     $local_fs $remote_fs
-# Default-Start:     2 3 4 5
+# Default-Start:     -
 # Default-Stop:      S 0 1 6
 # Short-Description: Startup/shutdown script for the Apache Traffic Server
 # Description:       Apache Traffic Server is fast, scalable and extensible
@@ -135,6 +144,9 @@
 # Ubuntu: Read configuration variable file if it is present
 test -r /etc/default/$NAME && . /etc/default/$NAME
 
+# RHEL/Fedora: Read configuration variable file if it is present
+test -r /etc/sysconfig/$NAME && . /etc/sysconfig/$NAME
+
 # Ubuntu: Load the VERBOSE setting and other rcS variables
 test -f /etc/default/rcS && . /etc/default/rcS
 
