diff -u netqmail-1.06/qmail-smtpd.c netqmail-1.06-new/qmail-smtpd.c
--- netqmail-1.06/qmail-smtpd.c	2020-01-16 05:39:21.575000010 +1100
+++ qmail-smtpd.c	2020-01-16 22:30:39.489000854 +1100
@@ -66,6 +66,8 @@
 int timeout = 1200;
 int maxrcpt = -1;
 unsigned int spfbehavior = 0;
+unsigned int require_auth = 0;
+int seenauth = 0;
 
 /* rejectrelaytest: start */
 unsigned int rejectrelaytest = 0;
@@ -136,6 +138,7 @@
   flush();
 }
 void straynewline() { qlogenvelope("rejected","badnewlines","","451"); logit("bad newlines"); out("451 See http://pobox.com/~djb/docs/smtplf.html.\r\n"); flush(); _exit(1); }
+void die_cannot_auth() { out("421 REQUIRE_AUTH set without valid AUTH program.\r\n"); flush(); _exit(1); }
 void die_pre_greet() { qlogenvelope("rejected","pregreet","","554"); out("554 SMTP protocol violation\r\n"); flush(); _exit(1); }
 
 void err_size() { qlogreceived("rejected","size","","552"); out("552 sorry, that message size exceeds my databytes limit (#5.3.4)\r\n"); }
@@ -168,6 +171,7 @@
 int err_authabrt() { out("501 auth exchange canceled (#5.0.0)\r\n"); return -1; }
 int err_input() { out("501 malformed auth input (#5.5.4)\r\n"); return -1; }
 void err_authfail() { qlogenvelope("rejected","authfailed","","535"); out("535 authentication failed (#5.7.1)\r\n"); }
+int err_authfirst() { out("503 AUTH first (#5.5.1)\r\n"); }
 void err_submission() { qlogenvelope("rejected","authrequired","","530"); out("530 Authorization required (#5.7.1) \r\n"); }
 void err_vrt() { qlogenvelope("rejected","validrcptto","","553"); out("553 sorry, this recipient is not in my validrcptto list (#5.7.1)\r\n"); }
 void die_brtlimit() { qlogenvelope("rejected","brtlimit","","421"); out("421 too many invalid addresses, goodbye (#4.3.0)\r\n"); flush(); _exit(1); }
@@ -976,7 +980,6 @@
 }
 /* rejectrelaytest: end */
 
-int seenauth = 0;
 int seenmail = 0;
 int rcptcount = 0;
 
@@ -1264,6 +1267,9 @@
 	break;
    }
 /* end chkuser code */
+
+    if (require_auth) if (!seenauth) { err_authfirst(); return; }
+
 /* authtlsvariables: start */
     /* if it is authenticated but MAIL FROM and AUTH USER are different */
     if (smtpauth && seenauth && forceauthmailfrom) {
@@ -2117,6 +2124,10 @@
   }
 }
 
+  x = env_get("REQUIRE_AUTH");
+  if (x) scan_ulong(x,&u); 
+  if (u>0) require_auth = 1;
+
 #ifdef TLS
 stralloc proto = {0};
 int ssl_verified = 0;

