diff -Naur mod_fastcgi-2.4.7_pure/fcgi_pm.c mod_fastcgi-2.4.7_mod-kill/fcgi_pm.c
--- mod_fastcgi-2.4.7_pure/fcgi_pm.c	2009-09-29 02:33:15.000000000 +0200
+++ mod_fastcgi-2.4.7_mod-kill/fcgi_pm.c	2014-09-29 15:22:24.000000000 +0200
@@ -133,7 +133,7 @@
         /* Send TERM to all processes */
         for (i = 0; i < numChildren; i++, proc++) 
         {
-            if (proc->state == FCGI_RUNNING_STATE) 
+            if (proc->state == FCGI_RUNNING_STATE || proc->state == FCGI_VICTIM_STATE)
             {
                 fcgi_kill(proc, SIGTERM);
             }
@@ -201,7 +201,7 @@
         /* Send KILL to all processes */
         for (i = 0; i < numChildren; i++, proc++) 
         {
-            if (proc->state == FCGI_RUNNING_STATE) 
+            if (proc->state == FCGI_RUNNING_STATE || proc->state == FCGI_VICTIM_STATE)
             {
                 fcgi_kill(proc, SIGKILL);
             }
